### DISCUSSION TITLE
Can you add an option to log the "raw" dynamic prompt used in a batch?

### DISCUSSION THREAD

**Discussion by keenerb** [adieyal/sd-dynamic-prompts#77] [Ideas]
So a simple example prompt:

(photo realistic), (highest quality),  (1 girl), (({blonde|redhead}))

With the Write prompts to file/Write all generated prompts to a file option selected, I get a filename of this:

photo-realistic-highest-quality-1-girl-blonderedhead.txt

That contains the post-dynamic prompt prompt:

(photo realistic), (highest quality),  (1 girl), ((redhead))

I'd like a way to log the full, original dynamic prompt that I can copy and paste and re-run.

**Comment by adieyal** [comment]
Good idea - will do

**Comment by adieyal** [comment]
This has been added in v1.1.0. The output file is now a csv, with positive and negative prompts as well as the raw prompts

**Comment by keenerb** [comment]
Fantastic, thanks!

**Comment by adieyal** [comment]
Could you please mark this question as answered? Otherwise it is hard for me to keep track of unanswered questions.

**Comment by adieyal** [comment]
You can now enable writing the raw prompt into the png metadata - there is an option in advanced settings.
